lib/mosaic-database/src/main/java/org/eclipse/mosaic/lib/database/spatial/WallFinder.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2022 Fraunhofer FOKUS and others. All rights reserved.
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contact: mosaic@fokus.fraunhofer.de
+ */
+
+package org.eclipse.mosaic.lib.database.spatial;
+
+import org.eclipse.mosaic.lib.database.Database;
+import org.eclipse.mosaic.lib.database.building.Building;
+import org.eclipse.mosaic.lib.database.building.Wall;
+import org.eclipse.mosaic.lib.math.Vector3d;
+import org.eclipse.mosaic.lib.spatial.Edge;
+import org.eclipse.mosaic.lib.spatial.KdTree;
+import org.eclipse.mosaic.lib.spatial.SpatialItemAdapter;
+import org.eclipse.mosaic.lib.spatial.SpatialTreeTraverser;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * This wall finder provides a quick spatial search to find all walls of buildings
+ * which can be found in the requested circular area.
+ */
+public class WallFinder {
+
+    private final KdTree<Edge<Vector3d>> wallTree;
+    private final SpatialTreeTraverser.InRadius<Edge<Vector3d>> wallSearch;
+
+    public WallFinder(Database db) {
+        List<Edge<Vector3d>> walls = new ArrayList<>();
+        for (Building building : db.getBuildings()) {
+            for (Wall wall : building.getWalls()) {
+                walls.add(new Edge<>(
+                        wall.getFromCorner().getCartesianPosition().toVector3d(),
+                        wall.getToCorner().getCartesianPosition().toVector3d()
+                ));
+            }
+        }
+        wallTree = new KdTree<>(new SpatialItemAdapter.EdgeAdapter<>(), walls);
+        wallSearch = new SpatialTreeTraverser.InRadius<>();
+    }
+
+    /**
+     * Finds and returns all walls in the vicinity of the given center point, which are
+     * within the given range.
+     *
+     * @param center the center point to search for walls
+     * @param range the maximum distance from the center points to search for walls
+     *
+     * @return a list of edges (pair of {@link Vector3d}) which represent the found walls
+     */
+    public Collection<Edge<Vector3d>> getWallsInRadius(Vector3d center, double range) {
```

Comment:
The only minor concern I have here is that we use the center coordinate of the walls to check if the wall is in range, as far as I can grasp it. This however, could lead to weird behaviour when only one point of the wall/edge is in viewing range.

Reply:
I think I fixed this by introducing an extension of `SpatialTreeTraverser.InRadius` which uses `Edge#getNearestPointOnEdge(Vector3d)` instead of `SpatialTreeTraverser#getCenter()`, see [1d36e1b](https://github.com/eclipse/mosaic/pull/252/commits/1d36e1bf4b0cdc2ae6a89da9b90cf3368018d6b8)